src/routing/approved.rs

```diff
@@ -1653,56 +1575,19 @@ impl Approved {
         self.relay_message(&msg)
     }
 
-    fn send_dkg_start(&mut self, new_elders_info: EldersInfo) -> Result<Vec<Command>> {
-        trace!("Send DKGStart for {}", new_elders_info);
+    fn send_dkg_start(&mut self, elders_info: EldersInfo) -> Result<Vec<Command>> {
+        trace!("Send DKGStart for {}", elders_info);
 
-        let dkg_key = DkgKey::new(&new_elders_info);
+        let dkg_key = DkgKey::new(&elders_info);
 
         // Send to all participants.
-        let recipients: Vec<_> = new_elders_info.elders.values().copied().collect();
+        let recipients: Vec<_> = elders_info.elders.values().copied().collect();
```

Comment:
this elders_info could be different to the real KenGen initialized internally, due to the retry attempt.

Reply:
A retry attempt doesn't change the elders info.

Reply:
ok. so the returned elders_info within the Error is the same one of the input.
got it.